doc/nrf/conf.py

```diff
@@ -103,18 +105,29 @@
 if kconfig_mapping:
     intersphinx_mapping["kconfig"] = kconfig_mapping
 
+# -- Options for doxyrunner plugin ---------------------------------------------
+
+doxyrunner_doxygen = os.environ.get("DOXYGEN_EXECUTABLE", "doxygen")
+doxyrunner_doxyfile = NRF_BASE / "doc" / "nrf" / "nrf.doxyfile.in"
+doxyrunner_outdir = NRF_BUILD / "doxygen"
+doxyrunner_fmt = True
+doxyrunner_fmt_vars = {
+    "NRF_BASE": str(NRF_BASE),
```

Comment:
Wouldn't it make more sense to keep doing that replacement with cmake, instead of repeating all required variables here?
Also, if we do it like this here, we should probably use a different file name than nrf.doxyfile.in, since that implies that you do the "normal" cmake replacement.

Reply:
This is how the plugin works, CMake doesn't take care of Doxygen anymore (unless we want to provide a standalone target, done e.g. in Zephyr). The technical discussions can be found here: https://github.com/zephyrproject-rtos/zephyr/pull/33934

Reply:
From what I understand, the plugin takes a doxyfile as input, and optionally does some replacements in the file before using it.
So why would we list the replacements manually if cmake can do them for free, and then we just provide the output doxyfile to the plugin?

Reply:
As mentioned we already discussed that. The concept behind is that Sphinx does not depend on CMake for operating, so one could potentially just invoke `sphinx-build` in the future.

Reply:
This is temporary until we add a `wet docbuild` command that will encapsulate this. The usage of cmake for building the doc will slowly fade away until it's replaced by west and sphinx itself.